.github/workflows/main.yml

```diff
@@ -22,6 +22,19 @@ jobs:
     - run: rustup component add rustfmt
     - run: cargo fmt --all -- --check
 
+  # Lint dependency graph for security advisories, duplicate versions, and
+  # incompatible licences
+  cargo_deny:
+    name: Cargo deny
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+    - uses: ./.github/actions/install-rust
+    - run: cargo install cargo-deny
```

Comment:
I'm compiling from crates.io here but alternatively I could download a precompiled binary from GitHub to make this run faster https://github.com/EmbarkStudios/cargo-deny/releases

This would mean that version upgrades require changing this file to get the newer binary, and there is no checksum to verify the binary with.

Reply:
I think using the precompiled binary would be good. We could perhaps embed the hash from the current release here?

Reply:
Oh yes of course! I'll do this

Reply:
I've just noticed that checksums are not checked for the other binaries being downloaded so I'll skip for now.